snapcraft/plugins/go.py

```diff
@@ -144,6 +144,15 @@ def _get_local_go_package(self):
             go_package = os.path.basename(os.path.abspath(self.options.source))
         return go_package
 
+    def _get_local_main_packages(self):
+        search_path = './{}/...'.format(self._get_local_go_package())
+        packages = self._run_output(['go', 'list', '-f',
+                                     '{{.ImportPath}} {{.Name}}',
+                                     search_path])
+        packages_split = [p.split() for p in packages.splitlines()]
+        main_packages = [p[0] for p in packages_split if p[1] == 'main']
+        return main_packages
+
```

Comment:
I like the use of -f here to get sensible output - do we care for a pretty error when the list is empty? This would right now result in "No such file or directory" which isn't necessarily clear.